src/Microsoft.Health.Dicom.Core/Features/Workitem/LoggingWorkitemStore.cs

```diff
@@ -92,5 +93,26 @@ public async Task<DicomDataset> GetWorkitemAsync(WorkitemInstanceIdentifier work
                 throw;
             }
         }
+
+        /// <inheritdoc />
+        public async Task DeleteWorkitemAsync(WorkitemInstanceIdentifier identifier, long? proposedWatermark = default, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(identifier, nameof(identifier));
+
+            LogQueryWorkitemDelegate(_logger, identifier, null);
+
+            try
+            {
+                await _workitemStore.DeleteWorkitemAsync(identifier, proposedWatermark, cancellationToken);
+
+                LogOperationSucceededDelegate(_logger, null);
+            }
+            catch (Exception ex)
+            {
+                LogOperationFailedDelegate(_logger, ex);
```

Comment:
We should get rid of this pattern, at least for new additions dont add delegate

Reply:
Done.